src/analysis/normed_space/pointwise.lean

```diff
@@ -95,13 +95,110 @@ by rw [vadd_ball, vadd_eq_add, add_zero]
 lemma vadd_closed_ball_zero (x : E) (r : ℝ) : x +ᵥ closed_ball 0 r = closed_ball x r :=
 by rw [vadd_closed_ball, vadd_eq_add, add_zero]
 
-variables [normed_space ℝ E]
+variables [normed_space ℝ E] {x y z : E} {δ ε : ℝ}
 
 /-- In a real normed space, the image of the unit ball under scalar multiplication by a positive
 constant `r` is the ball of radius `r`. -/
 lemma smul_unit_ball_of_pos {r : ℝ} (hr : 0 < r) : r • ball 0 1 = ball (0 : E) r :=
 by rw [smul_unit_ball hr.ne', real.norm_of_nonneg hr.le]
 
+-- This is also true for `ℚ`-normed spaces
+lemma exists_dist_eq (x z : E) {a b : ℝ} (ha : 0 ≤ a) (hb : 0 ≤ b) (hab : a + b = 1) :
+  ∃ y, dist x y = b * dist x z ∧ dist y z = a * dist x z :=
+begin
+  refine ⟨a • x + b • z, _, _⟩; simp_rw dist_eq_norm,
+  calc _ = ∥(a • x - a • x) + b • (x - z)∥
+    : by rw [smul_sub, sub_add_sub_comm, convex.combo_self hab]
+  ... = _ : by rw [sub_self, zero_add, norm_smul_of_nonneg hb],
+  calc _ = ∥a • (x - z) + (b • z - b • z)∥
+    : by rw [smul_sub, sub_add_sub_comm, convex.combo_self hab]
+  ... = _ : by rw [sub_self, add_zero, norm_smul_of_nonneg ha],
+end
+
+lemma exists_dist_le_le (hδ : 0 ≤ δ) (hε : 0 ≤ ε) (h : dist x z ≤ ε + δ) :
+  ∃ y, dist x y ≤ δ ∧ dist y z ≤ ε :=
+begin
+  obtain rfl | hε' := hε.eq_or_lt,
+  { exact ⟨z, by rwa zero_add at h, (dist_self _).le⟩ },
+  have hεδ := add_pos_of_pos_of_nonneg hε' hδ,
+  refine (exists_dist_eq x z (div_nonneg hε $ add_nonneg hε hδ) (div_nonneg hδ $ add_nonneg hε hδ) $
+    by rw [←add_div, div_self hεδ.ne']).imp (λ y hy, _),
+  rw [hy.1, hy.2, div_mul_comm', div_mul_comm' ε],
+  rw ←div_le_one hεδ at h,
+  exact ⟨mul_le_of_le_one_left hδ h, mul_le_of_le_one_left hε h⟩,
+end
+
+-- This is also true for `ℚ`-normed spaces
+lemma exists_dist_le_lt (hδ : 0 ≤ δ) (hε : 0 < ε) (h : dist x z < ε + δ) :
+  ∃ y, dist x y ≤ δ ∧ dist y z < ε :=
+begin
+  refine (exists_dist_eq x z (div_nonneg hε.le $ add_nonneg hε.le hδ)
+    (div_nonneg hδ $ add_nonneg hε.le hδ) $
+    by rw [←add_div, div_self (add_pos_of_pos_of_nonneg hε hδ).ne']).imp (λ y hy, _),
+  rw [hy.1, hy.2, div_mul_comm', div_mul_comm' ε],
+  rw ←div_lt_one (add_pos_of_pos_of_nonneg hε hδ) at h,
+  exact ⟨mul_le_of_le_one_left hδ h.le, mul_lt_of_lt_one_left hε h⟩,
+end
```

Comment:
```suggestion
lemma exists_dist_le_lt (hδ : 0 ≤ δ) (hε : 0 < ε) (h : dist x z < ε + δ) :
  ∃ y, dist x y ≤ δ ∧ dist y z < ε :=
begin
  obtain ⟨y, yz, xy⟩ := exists_dist_lt_le hε hδ
    (show dist z x < δ + ε, by simpa [dist_comm, add_comm] using h),
  exact ⟨y, by simp [dist_comm x y, dist_comm y z, *]⟩,
end
```
if you want to condense a bit, you can move this one below `exists_dist_lt_le` and use commutativity